dev/io.openliberty.microprofile.openapi.2.0.internal_fat/fat/src/io/openliberty/microprofile/openapi20/fat/deployments/MergeConfigServerXMLTest.java

```diff
@@ -63,18 +68,295 @@ public class MergeConfigServerXMLTest {
+        WebArchive war2 = ShrinkWrap.create(WebArchive.class, "test2.war")
+                                    .addClasses(DeploymentTestApp.class, DeploymentTestResource.class);
+        deployApp(war2);
+
+        // Download the documentation
+        String doc = OpenAPIConnection.openAPIDocsConnection(server, false).download();
+        JsonNode openapiNode = OpenAPITestUtil.readYamlTree(doc);
+
+        if (OpenAPITestUtil.getOpenAPIFeatureVersion() < 4.0f) {
+            // Default is first module only
+            // check that documentation includes only app 1
+            OpenAPITestUtil.checkPaths(openapiNode, 1, "/test");
+
+            // Test that a message was output
+            assertThat(server.findStringsInLogs(" I CWWKO1663I:.*Combining OpenAPI documentation from multiple modules is disabled.*test1"),
+                       hasSize(1));
+
+            // remove app 1
+            removeApp(war1);
+
+            // check that documentation includes only app 2
+            doc = OpenAPIConnection.openAPIDocsConnection(server, false).download();
+            openapiNode = OpenAPITestUtil.readYamlTree(doc);
+            OpenAPITestUtil.checkPaths(openapiNode, 1, "/test");
+        } else {
+            // Default is all modules
+            // Check that documentation includes both app 1 and app 2
+            OpenAPITestUtil.checkPaths(openapiNode, 2, "/test1/test", "/test2/test");
+
+            // Check there's no "first module only" message
+            assertThat(server.findStringsInLogs("CWWKO1663I"),
+                       hasSize(0));
+        }
+    }
+
+    @Test
+    public void testFirstModuleOnly() throws Exception {
+        setMergeConfig(list("first"), null, null);
+
+        // deploy app 1
+        WebArchive war1 = ShrinkWrap.create(WebArchive.class, "test1.war")
+                                    .addClasses(DeploymentTestApp.class, DeploymentTestResource.class);
+        deployApp(war1);
+
+        // deploy app 2
+        WebArchive war2 = ShrinkWrap.create(WebArchive.class, "test2.war")
+                                    .addClasses(DeploymentTestApp.class, DeploymentTestResource2.class);
+        deployApp(war2);
+
+        // check that documentation includes only app 1
+        String doc = OpenAPIConnection.openAPIDocsConnection(server, false).download();
+        JsonNode openapiNode = OpenAPITestUtil.readYamlTree(doc);
+        OpenAPITestUtil.checkPaths(openapiNode, 1, "/test");
+
+        // remove app 1
+        removeApp(war1);
+
+        // check that documentation includes only app 2
+        doc = OpenAPIConnection.openAPIDocsConnection(server, false).download();
+        openapiNode = OpenAPITestUtil.readYamlTree(doc);
+        OpenAPITestUtil.checkPaths(openapiNode, 1, "/test2");
+
+        // Test that merging disabled message was output (at some point)
+        assertThat(server.findStringsInLogs(" I CWWKO1663I:.*Combining OpenAPI documentation from multiple modules is disabled."),
```

Comment:
The equivalent line in MergeConfigTest ends in `.*test1` should it here?

Reply:
The error message here should include the application name. However, the error message is only output once for each server start (this message appears when you deploy an additional app and we don't want to spam it if the user deploys lots of apps).

In `MergeConfigTest` the server is restarted for each test so each test will get its own instance of the message. In this test the server is started once for the whole class, so the message is only output when the first test that prompts it runs.

Since we don't know the test order, we don't know what application name will be in the message. When the second test runs, it's finding the message that was output when the first test ran.